enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/grpc/HealthService.java

```diff
@@ -24,4 +69,36 @@ public void check(HealthCheckRequest request, StreamObserver<HealthCheckResponse
         responseObserver.onNext(response);
         responseObserver.onCompleted();
     }
+
+    /**
+     * Check the health of the discovery clients.
+     *
+     * @param responseObserver the response observer
+     * @return true if all the discovery clients are healthy, false otherwise
+     */
+    private static boolean checkDiscoveryClientsHealth(StreamObserver<HealthCheckResponse> responseObserver) {
+        if (!isDiscoveryClientsInitialFetchCompleted) {
+            logger.info("Checking health of discovery clients");
+            boolean initialized = true;
+            for (String clientName : discoveryClients.keySet()) {
+                DiscoveryClient discoveryClient = discoveryClients.get(clientName);
+                if (!discoveryClient.isInitialFetchCompleted()) {
+                    logger.info("Discovery client: {} has not completed initial fetch", clientName);
+                    initialized = false;
+                    break;
+                } else {
+                    logger.debug("Discovery client: {} has completed initial fetch", clientName);
+                }
+            }
+            isDiscoveryClientsInitialFetchCompleted = initialized;
```

Comment:
Shall we put an info log here?
```suggestion
            isDiscoveryClientsInitialFetchCompleted = initialized;
            logger.info("All discovery clients are healthy");
```